examples/08_interpolation_join.py

```diff
@@ -0,0 +1,179 @@
+# Loading the flights table
+# -------------------------
+# We load the flights table and join it to the airports table using the flights’ "Origin" which refers to the departure airport’s IATA code.
+# We use only a subset to speed up the example.
+
+flights = fetch_figshare("41771418").X[["Year_Month_DayofMonth", "Origin", "ArrDelay"]]
+flights = flights.sample(20_000, random_state=0, ignore_index=True)
+airports = fetch_figshare("41710257").X.loc[
+    :, ["iata", "airport", "state", "lat", "long"]
+]
+flights = flights.merge(airports, left_on="Origin", right_on="iata")
+# printing the first row is more readable than the head() when we have many columns
+flights.iloc[0]
+
+######################################################################
+# Joining the flights and weather data
+# ------------------------------------
+# As before, we initialize our join transformer with the weather table.
+# Then, we use it to transform the flights table -- it adds a "TMAX" column containing the predicted maximum daily temperature.
+#
+
+joiner = InterpolationJoiner(
+    aux_table,
+    main_key=["lat", "long", "Year_Month_DayofMonth"],
+    aux_key=["LATITUDE", "LONGITUDE", "YEAR/MONTH/DAY"],
+)
+join = joiner.fit_transform(flights)
+join.head()
+
+######################################################################
+# Sanity checks
+# -------------
+# This time we do not have a ground truth for the temperatures.
+# We can perform a few basic sanity checks.
+
+state_temperatures = join.groupby("state")["TMAX"].mean().sort_values()
+
+######################################################################
+# states with the lowest average predicted temperatures: Alaska, Montana, North Dakota, Washington, Minnesota
+state_temperatures.head()
+
+######################################################################
+# states with the highest predicted temperatures: Puerto Rico, Virgin Islands, Hawaii, Florida, Louisiana
+state_temperatures.tail()
+
+######################################################################
+# Higher latitudes (farther up north) are colder -- the airports in this dataset are in the United States.
+fig, ax = plt.subplots()
+ax.scatter(join["lat"], join["TMAX"])
+ax.set_xlabel("Latitude (higher is farther north)")
+ax.set_ylabel("TMAX")
+
+######################################################################
+# Winter months are colder than spring -- in the north hemisphere January is colder than April
+#
+
+import seaborn as sns
+
+join["month"] = join["Year_Month_DayofMonth"].dt.strftime("%m %B")
+plt.figure()
+sns.barplot(data=join.sort_values(by="month"), y="month", x="TMAX")
+
+######################################################################
+# Of course these checks do not guarantee that the inferred values in our ``join`` table’s ``TMAX`` column are accurate.
+# But at least the ``InterpolationJoiner`` seems to have learned a few reasonable trends from its training table.
+
```

Comment:
Did you try predicting the flights delay with the new interpolated table? How do the results compare to a simple fuzzy join ?

Reply:
I didn't because I felt the example was quite long already, but I can if you think it's useful! I'd be quite surprised if it noticeably improved delay predictions though

Reply:
Ok, I guess it's pertinent to add only if there are noticeable improvements in prediction to motivate the usage, otherwise no need to prolong the example..

Reply:
I compared the r2 scores by adding this to the end of the example:

```python
from sklearn.model_selection import cross_val_score
from sklearn.ensemble import HistGradientBoostingRegressor
from sklearn.pipeline import make_pipeline
from skrub import TableVectorizer

flights = flights.dropna(subset="ArrDelay")
join = join.dropna(subset="ArrDelay")
X = flights.drop("ArrDelay", axis=1)
y = flights["ArrDelay"].values
regressor = make_pipeline(TableVectorizer(), HistGradientBoostingRegressor())
score = cross_val_score(regressor, X, y)
print(score)
X = join.drop("ArrDelay", axis=1)
score = cross_val_score(regressor, X, y)
print(score)
```

in both cases the prediction is at chance level (probably because I kept very few features from the flights dataset). it makes the example run slower. I guess we can keep it out for now and maybe search for a dataset where the benefit of the interpolation joiner on the downstream task is more obvious?

Reply:
Yes, let's keep it out then. Thanks for checking.